syntax = "proto3";

package keyvaluestorage;

import "kvstore/keyvalue.proto";

service KeyValueStorage {
  rpc Put (PutRequest) returns (PutResponse) {}
  rpc Get (GetRequest) returns (GetResponse) {}
} 

message PutRequest {
  Value value = 1;
}

message PutResponse {
  bool success = 1;
  Key key = 2;
}

message GetRequest {
  Key key = 1;
}

message GetResponse {
  bool present = 1;
  Value value = 2;
}

